src/status_im/multiaccounts/core.cljs

```diff
@@ -12,84 +11,35 @@
             [status-im.utils.gfycat.core :as gfycat]
             [status-im.utils.identicon :as identicon]
-            [status-im.utils.utils :as utils]
             [status-im2.setup.hot-reload :as hot-reload]
             [status-im2.common.theme.core :as utils.theme]
             [taoensso.timbre :as log]
-            [status-im2.contexts.shell.animation :as shell.animation]))
+            [status-im2.contexts.shell.animation :as shell.animation]
+            [status-im.contact.db :as contact.db]))
 
 ;; validate that the given mnemonic was generated from Status Dictionary
 (re-frame/reg-fx
  ::validate-mnemonic
  (fn [[passphrase callback]]
    (native-module/validate-mnemonic passphrase callback)))
 
-(defn contact-names
-  "Returns map of all existing names for contact"
-  [{:keys [name
-           display-name
-           preferred-name
-           alias
-           public-key
-           ens-verified
-           nickname]}]
-  (let [ens-name (or preferred-name
-                     name)]
-    (cond-> {:nickname         nickname
-             :display-name     display-name
-             :three-words-name (or alias (gfycat/generate-gfy public-key))}
-      ;; Preferred name is our own otherwise we make sure it's verified
-      (or preferred-name (and ens-verified name))
-      (assoc :ens-name (or (stateofus/username ens-name) ens-name)))))
-
-(defn contact-two-names
-  "Returns vector of two names in next order nickname, ens name, display-name, three word name, public key"
-  [{:keys [names
-           compressed-key
-           public-key]
-    :as   contact} public-key?]
-  (let [{:keys [nickname
-                ens-name
-                display-name
-                three-words-name]}
-        (or names (contact-names contact))
-        non-empty-names (remove string/blank? [nickname ens-name display-name three-words-name])]
-    (if (> (count non-empty-names) 1)
-      (vec (take 2 non-empty-names))
-      [(first non-empty-names)
-       (when public-key? (utils/get-shortened-address (or compressed-key public-key)))])))
-
-(defn contact-with-names
-  "Returns contact with :names map "
-  [contact]
-  (let [contact' (assoc contact :names (contact-names contact))]
-    (assoc contact' :two-names (contact-two-names contact' true))))
-
 (defn displayed-name
-  "Use preferred name, display-name, name or alias in that order"
-  [{:keys [name display-name preferred-name alias public-key ens-verified]}]
-  (let [display-name (if (string/blank? display-name) nil display-name)
-        ens-name     (or preferred-name
-                         display-name
-                         name)]
-    ;; Preferred name is our own otherwise we make sure it's verified
-    (if (or preferred-name (and ens-verified name))
-      (let [username (stateofus/username ens-name)]
-        (or username ens-name))
-      (or display-name alias (gfycat/generate-gfy public-key)))))
+  "Legacy"
+  [{:keys [first-name]}]
```

Comment:
oh wow, took a minute to understand :)

Reply:
it's not exactly the same, since the arity of the function is different, and sometimes that matters, for example:

```
(def displayed-name-1 :first-name)

(def displayed-name-2 [{:keys [first-name]}] first-name)
```
will give different result in cases like:

```
(displayed-name-1 {} "b") -> returns `b`
(displayed-name-2 {} "b") -> returns `nil`
```

and that's why we don't use it in `reg-sub`, where you might want to do

```
(re-frame/reg-sub
 :activity-center-state
 :field-on-state)
```

which looks neat, but you get the event name instead if the `activity-center-state` does not have the key, fun fact :)